### PR TITLE
Add in leader elected and lost quorum advisories.

### PR DESCRIPTION
**Pull request by derekcollison**
/cc @nats-io/core

### CODE REVIEW

File: server/jetstream_cluster.go [hunk 1]
```diff
@@ -1241,21 +1241,67 @@ func (js *jetStream) applyStreamEntries(mset *Stream, ce *CommittedEntry) (bool,
...
+		if node := mset.raftNode(); node != nil && !node.Quorum() {
+			s.Warnf("JetStream cluster stream '%s > %s' has lost quorum, stalled.", sa.Client.Account, stream)
+			s.publishAdvisory(mset.account(), JSAdvisoryStreamQuorumLostPre+"."+stream, &JSStreamQuorumLostAdvisory{
+				TypedEvent: TypedEvent{
+					Type: JSStreamQuorumLostAdvisoryType,
+					ID:   nuid.Next(),
+					Time: time.Now().UTC(),
+				},
+				Stream:   stream,
+				State:    mset.State(),
```
Comment (ripienaar) [inline server/jetstream_cluster.go:1293]: without consumer state i wonder if its worth having this really? I am not sure what I'll do with this info if I had it, what did you have in mind as a usecase?

**Review by ripienaar — APPROVED** [review]
Just a q, otherwise, LGTM

File: server/jetstream_cluster.go (hunk 1 above)
Comment (derekcollison) [reply server/jetstream_cluster.go:1293]: What would you do with state here?

**Comment by derekcollison** [comment]
FYI this approach is a bit more fragile than I want, requires old leader to not be the one lost to lose quorum. Have an idea I am going to try to make it more robust. Will update PR if it works.

File: server/jetstream_cluster.go (hunk 1 above)
Comment (ripienaar) [reply server/jetstream_cluster.go:1293]: I can't think of anything I'd do with it :)

**Review by ripienaar — APPROVED** [review]
LGTM

**Pull request merged**